File: src/game/tools/rpc/output/wrap/IntArrayWrap.go
package wraps

import (
	rpc "tipcat.com/common/rpc/simple"
)

type IntArrayWrap struct {
	
	Value []int32 // 整型数组
	rpc.Wrapper
}

func (w *IntArrayWrap)Decode(pck *rpc.Packet) rpc.Wrapper {
	
	w.Value = make([]int32, int(pck.PopInt16()))
	for i := 0; i < len(w.Value); i++ {
		w.Value[i] = pck.PopInt32()
	}
	
	return w
}

func (w *IntArrayWrap)Encode(pck *rpc.Packet) {
	
	if w.Value == nil {
		pck.PutInt16(0)
	} else {
    	pck.PutInt16(int16(len(w.Value)))
    	for i := 0; i < len(w.Value); i++ {
    		pck.PutInt32(w.Value[i])
    	}
    }
	
}

File: src/game/tools/rpc/output/wrap/IntArrayWrap.cs
using System;
using Common.Net.Simple;

namespace Game 
{
	public class IntArrayWrap : IWrapper
	{	
		public int[] Value; // 整型数组
		
		public void Decode(Packet pck)
		{	
			Value = new int[pck.GetShort()];
			for (int i = 0; i < Value.Length; i++)
			{
				Value[i] = pck.GetInt();
			}
			
		}
		
        public void Encode(Packet pck)
        {	
        	if (Value == null) pck.PutShort((short)0); 
        	else
        	{
	        	pck.PutShort((short)Value.Length);
	        	for(int i = 0; i < Value.Length; i++)
	        	{
	        		pck.PutInt(Value[i]);
	        	}
	        }
        	
        }
	}
}